src/init.cpp

```diff
@@ -1701,6 +1701,13 @@ bool AppInitMain(NodeContext& node, interfaces::BlockAndHeaderTipInfo* tip_info)
         if (index == std::string::npos) {
             if (Lookup(bind_arg, bind_addr, GetListenPort(), false)) {
                 connOptions.vBinds.push_back(bind_addr);
+                if (IsBadPort(bind_addr.GetPort())) {
+                    InitWarning(strprintf(
+                        _("-bind request to listen on port %u. This port is considered \"bad\" and "
```

Comment:
I think that would be an overkill. It would require getting the service name from the port somehow, like in your suggestion below. But then `IsBadPort()` would not be a suitable name anymore, it should be renamed to something like `GetBadPortName()` which would imply that the passed value is bad. Maybe `IsBadPortAndIfYesThenReturnItsName()` :-( I would Keep It Simple, Stupid. The names are now in `doc/p2p-bad-ports.md` which is mentioned in this message.

Reply:
Agree that that would be overkill. The ports are well-known and if not it's only a web search away.